src/foam/nanos/approval/ApprovalRequest.js

```diff
@@ -795,5 +775,67 @@
       },
       tableWidth: 100
     }
+  ],
+  listeners: [
+    {
+      name: 'approveWithMemo',
+      code: function(memo) {
+        var approvedApprovalRequest = this.clone();
+        approvedApprovalRequest.status = this.ApprovalStatus.APPROVED;
+        approvedApprovalRequest.memo = memo;
+
+        this.approvalRequestDAO.put(approvedApprovalRequest).then((req) => {
+          this.approvalRequestDAO.cmd(this.AbstractDAO.RESET_CMD);
+          this.finished.pub();
+          this.notify(this.SUCCESS_APPROVED, '', this.LogLevel.INFO, true);
+
+          if ( this.subStack.top.length > 0 ) {
```

Comment:
Do we need this 'if'? The back() method in stack already has:
      isEnabled: function(pos) { return pos > 0; },